File: src/main/ioke/lang/Builtin.java
/*
 * See LICENSE file in distribution for copyright and licensing information.
 */
package ioke.lang;

import ioke.lang.exceptions.ControlFlow;

/**
 *
 * @author <a href="mailto:ola.bini@gmail.com">Ola Bini</a>
 */
public abstract class Builtin {
    public abstract Object load(Runtime runtime, IokeObject context, IokeObject message) throws ControlFlow;
}// Builtin

File: src/ikc/main/Ioke.Lang/Builtin.cs
namespace Ioke.Lang {
    public abstract class Builtin {
        public class Delegate : Builtin {
            public delegate object LoadDelegate (Runtime runtime, IokeObject context, IokeObject message);
            
            LoadDelegate ld;
            public Delegate(LoadDelegate ld) {
                this.ld = ld;
            }

            public override object Load(Runtime runtime, IokeObject context, IokeObject message) {
                return ld(runtime, context, message);
            }
        }

        public abstract object Load(Runtime runtime, IokeObject context, IokeObject message);
    }
}
